src/pages/PathSelection.test.js

```diff
@@ -129,4 +135,25 @@ describe('PathSelection Component UI Tests', () => {
       ).toBeInTheDocument();
     });
   });
+
+  it('should retrieve all paths from the database', async () => {
+    const { container } = render(
+      <BrowserRouter>
+        <PathSelection />
+      </BrowserRouter>
+    );
+
+    await waitFor(() => {
```

Comment:
Miksi tämä (ja samalla huomasin että muissakin testeissä) on wräpätty tähän await waitFor lohkoon?
Heitin sen sisään ihan simppelin console.log() komennon, se pyörittää sitä useita kertoja ja failaa koko testi casen. Oletko varma ettei tuo vain peitä failaavia testejä, mutta se ei oikeasti testaa tuota toiminnallisuutta?

Reply:
No tästä mä en osaa ihan sanoa, katoin muista jeesii, mut siis ajattelin sen niin et tollee se oottaa ne polut näkyviin? Voi olla et väärä ajatus, mut mun mielestä muuten se ei löytänyt sitä polkuu. Mut siis ei mul tälle enempää oo annettavaa oikein...

Reply:
Ja oon käyttänyt tota waitForia aikasemminki esim. GameMenun testeissä (niissä näköjään toimis vissiin ilmankin). Onks se sielläkin sitten virhe? En oikeen saa kiinni, et mikä siitä tekee virheen, et ehkä sit täytyy vaa jättää tää ja kattoo joskus porukalla...

Reply:
Itse koitin myös hetken aikaa tuota fixailla, mutta ei tää test framework oo tarpeeks tuttu selkeesti :D Katsotaan porukalla vaikka, itsekin haluaisin saada tietää miten tää flow pitäis toimia. Jätetään tää PR toistaseks auki, ei oo mikään kriittinen muutenkaan.

Reply:
Tappelin tän tyylisten testien kans aiemmin tällä viikolla mut taisin keksiä tavan miten noita db funktioita voi mockata ja et saa polkuja näkyviin! Tuolta #115 PR:stä voi käydä kurkkaamassa, kun todennäköisesti ne auttais tässäkin. Toki sillä varauksella et näitä ei oo vielä katselmoitu mut menee ainakin testit läpi ja tsekkasin ettei toi waitFor pyöri ikuisuutta.

Reply:
Jeps, tuo Neeran PR:n testaus tapa vaikuttaa juuri siltä mitä hain. Tietokanta mockataan ja testin alussa kerrotaan mitä kannasta tulee ja testataan miten ohjelma reagoi eri tilanteisiin. Muokkaa nämä testit käyttämään tuota samaa tyyliä.